### ISSUE TITLE
Test marked as `successful` on actual failure

### ISSUE THREAD

**Issue by tkonieczny** [kubeshop/testkube#3023]
**Describe the bug**
Test execution failed on init, but the test is marked as successful.

**To Reproduce**
1. Create a Cypress test with `git-file`:
```
apiVersion: tests.testkube.io/v3
kind: Test
metadata:
  name: cypress-git-file-test-1
  namespace: testkube
spec:
  type: cypress/project
  content:
    type: git-file
    repository:
      type: git-file
      uri: https://github.com/kubeshop/testkube.git
      path: test/cypress/executor-smoke/cypress-11
```
2. Run the test
3. Check the error message and test result
![Screenshot_20230112_135347](https://user-images.githubusercontent.com/1473096/212071524-55f3e50d-9ab1-4de8-b3ef-1da0b3514480.png)

**Expected behavior**
The test that failed on any stage should never be marked as success.
`git-file` shouldn't be used in case of Cypress, but that's something for another issue.

**Comment by tkonieczny** [comment]
Looks like the result is "random" - this test failed twice with exactly the same error message. Once the result was `fail`, the second one `success`.
![Screenshot_20230113_125554](https://user-images.githubusercontent.com/1473096/212315016-d1b05f74-341c-4e60-ad18-1aa51ff37830.png)

**Comment by vLia** [comment]
Hey @tkonieczny, a fix was released in Testkube [v1.8.28](https://github.com/kubeshop/testkube/releases/tag/v1.8.28), testing it on demo shows the issue is gone.

Please let me know if there are any further issues with this on the latest Testkube release.
